### PR TITLE
Fix for crashing lambdas in a package [TG-2478]

### PR DESCRIPTION
**Pull request by majakusber**
When the class including a lambda function was in a package we observed a crash. The fix is to format the class name properly when parsing the lambda method handle.

### CODE REVIEW

File: src/java_bytecode/java_bytecode_parser.cpp [hunk 1]
```diff
@@ -1777,9 +1777,13 @@ java_bytecode_parsert::parse_method_handle(const method_handle_infot &entry)
   const name_and_type_infot &name_and_type =
     ref_entry.get_name_and_type(pool_entry_lambda);
 
+  std::string class_name = class_entry.get_name(pool_entry_lambda);
```
Comment (mgudemann) [inline src/java_bytecode/java_bytecode_parser.cpp:1780]: would you mind adding an example here what happens here and why this is required?

**Review by mgudemann — APPROVED** [review]

File: src/java_bytecode/java_bytecode_parser.cpp (hunk 1 above)
Comment (majakusber) [reply src/java_bytecode/java_bytecode_parser.cpp:1780]: Do the comments below this line make it a bit clearer perhaps? We need to replace dots by dollars (formatting for inner classes) and then slashes by dots (formatting for packages).

Comment (mgudemann) [reply src/java_bytecode/java_bytecode_parser.cpp:1780]: yes, it's ok, I was just wondering about the `$` in lambda function names, so I wondered whether we are sure here not to replace those.

**Review by mgudemann — APPROVED** [review]

File: src/java_bytecode/java_bytecode_parser.cpp (hunk 1 above)
Comment (majakusber) [reply src/java_bytecode/java_bytecode_parser.cpp:1780]: Yes, that's a good point. They are not being replaced here, only the class name is manipulated, the lambda name is appended to the string afterwards.

**Review by thk123 — APPROVED** [review]

**Comment by thk123** [comment]
Mac builds not currently starting on Travis due to ongoing issue -- all other CI checks have passed.

**Pull request merged**